### PR TITLE
Laboratory work #3, Pareshina Evgeniya - 20FPL2 

### PR DESCRIPTION
**Pull request by eapareshina**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -31,38 +62,62 @@ class LetterStorage:
...
         :param letter: a letter
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str) or not letter:
+            return -1
+
+        if letter not in self.storage:
+            self.storage[letter] = self.unique_id
+            self.unique_id += 1
```
Comment (dmitry-uraev) [inline lab_3/main.py:78]: can you assign ids without adding additional self.unique_id parameter?

File: lab_3/main.py [hunk 2]
```diff
@@ -31,38 +62,62 @@ class LetterStorage:
...
 
     def get_letter_by_id(self, letter_id: int) ->str or int:
         """
         Gets a letter by a unique id
         :param letter_id: a unique id
         :return: letter
         """
-        pass
+        if not (isinstance(letter_id, int)
+                and letter_id in self.storage.values()):
```
Comment (dmitry-uraev) [inline lab_3/main.py:99]: Does this one works?

File: lab_3/main.py [hunk 3]
```diff
@@ -31,38 +62,62 @@ class LetterStorage:
...
-        pass
+
+        if not isinstance(corpus, tuple):
+            return -1
+
+        for sentence in corpus:
+            for word in sentence:
+                for letter in word:
+                    self._put_letter(letter)
+        return 0
```
Comment (dmitry-uraev) [inline lab_3/main.py:120]: good

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good start! Waiting the implementation for higher scores

File: lab_3/main.py (hunk 2 above)
Comment (eapareshina) [reply lab_3/main.py:99]: it works, all tests are passed

File: lab_3/main.py (hunk 1 above)
Comment (eapareshina) [reply lab_3/main.py:78]: okay, i'll rewrite this part

Comment (eapareshina) [reply lab_3/main.py:78]: I think I can use len() in order to avoid unique_id
self.storage[letter] = len(self.storage) + 1

**Pull request closed without merging**